README.md

```diff
@@ -99,15 +100,132 @@ use, and what the relevant image actually builds.
+## Issues & further building instructions
 
 If you run into trouble building, please consult
 the
 [GHC Building Guide](https://ghc.haskell.org/trac/ghc/wiki/Building)
 for possible answers.
 
-## Filing bugs
 
-Please note that this is under active development, but if you find
-something that definitiely looks off you are welcome to submit a bug
-through GitHub.
+Usage documentation
+===================
+
+The extention is provided as-is, but here is some basic documentation of how
+to use it.
+
+
+## Activating the extension
+The extension is _always on_ for the time being, so no need to set any
+flags or pragmas. In the future it will be an ordinary extension.
+
+
+## Writing linear functions
+Only the unicode lollipop arrow `⊸` is currently recognised. In the future there
+will be an ASCII version, but no decision has been taken on notation.
+
+When writing a function, you declare in the types which arguments are to be
+checked for linearity. If you do not treat these as such in your
+implementation, the type checker will nag on you:
+
+```
+λ> let frugal :: a ⊸ (a,a); frugal a = (a,a)
+
+<interactive>:2:33: error:
+    • Couldn't match expected weight ‘1’ of variable ‘a’ with actual weight ‘ω’
+    • In an equation for ‘frugal’: frugal a = (a, a)
+
+
+λ> let wasteful :: a ⊸ b ⊸ a; wasteful a b = a
+
+<interactive>:3:39: error:
+    • Couldn't match expected weight ‘1’ of variable ‘b’ with actual weight ‘0’
+    • In an equation for ‘wasteful’: wasteful a b = a
+```
+
+As you can see, the type checker gives some hints on _how_ you treated your
+variable wrongly, multiplicity `ω` means it was used more than once, or a 
+varying number in different code branches. 
+
+This `expected ‘1’ but actually ‘ω’`-error also occurs when the variables are
+passed to unrestricted functions, this is not always very obvious so be
+mindful of that possibility: 
+
+```
+λ> let plus1 :: Int ⊸ Int; plus1 x = x + 1
+
+<interactive>:12:31: error:
+    • Couldn't match expected weight ‘1’ of variable ‘x’ with actual weight ‘ω’
+    • In an equation for ‘plus1’: plus1 x = x + 1
+```
+
+This unfortunately applies to _all of prelude_ (`+` is the offender
+here), which means you might see this a few times.
+
+
+Currently the type checker does not check linearity when not asked to do so,
+hence it is a requirement to explicitly declare the types of linear functions.
+Otherwise, they are assumed to be unrestricted which can give you errors from
+using functions that are _implemented_ linearly, but not declared as such.
+In particular, functions in `where`-clauses are easy to forget to annotate
+(`-XPartialTypeSignatures` or `-XScopedTypeVariables` are useful for this),
+but lambdas are inferred properly.
+
+
+## Calling them
+By design, there is no obligations put on the caller of a linear function. This
+means any linear function can take unrestricted arguments, which is great
+since it makes all functions "linearly polymorphic". However, passing a
```

Comment:
I don't think we need to say it's "great", and not all function, only first order linear functions are "polymorphic" in this sense.